react/components/RenderProvider.tsx

```diff
@@ -544,14 +548,19 @@ export class RenderProvider extends Component<
       runtime: { rootPath },
     } = this.props
     const { pages } = this.state
-    options.rootPath = rootPath
-    options.modifiers = this.navigationRouteModifiers
 
     this.navigationModifierOptions = {
       ...this.navigationModifierOptions,
       ...options.modifiersOptions,
     }
-    options.modifiersOptions = this.navigationModifierOptions
+
+    options = {
+      ...options,
```

Comment:
I think it's better to keep how it was, without destructuring. I'm afraid of the side effects this might cause, performance-wise and behaviour-wise